components/bl-carousel-component/src/index.js

```diff
@@ -0,0 +1,137 @@
+import { useState, useEffect, useRef } from 'react';
+import { NextButton, PrevButton, List, CarouselIndicators } from './subcomponents';
+import { getPrevImg, getNextImg } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Carousel({ component, eventHandlers }) {
+  const {
+    display,
+    classList,
+    imagesData,
+    heightImage,
+    autoplayDelay,
+    animationDuration,
+    withControls,
+    withIndicators,
+    animationType
+  } = component;
+  const { onNextButton, onPrevButton, onMouseEnter, onMouseLeave } = eventHandlers;
+
+  const [data, setData] = useState(imagesData);
+  const [currentImg, setCurrentImg] = useState(0);
+  const [isNextAnimation, setIsNextAnimation] = useState(false);
+  const [isPrevAnimation, setIsPrevAnimation] = useState(false);
+  const [nextCurrentImage, setNextCurrentImage] = useState(0);
+  const [isAutoplay, setIsAutoplay] = useState(true);
+  const autoplayId = useRef();
+
+  const nextImg = getNextImg(currentImg, data.length);
```

Comment:
const { nextImg, prevImage } = getAdjacentImages(currentImg, data.length);

getAdjacentImages - returns both indexes for prev and next.